src/query/functions/binary/base.go

```diff
@@ -142,6 +147,14 @@ func (n *baseNode) Process(queryCtx *models.QueryContext, ID parser.NodeID, b bl
 	return n.controller.Process(queryCtx, nextBlock)
 }
 
+func (n *baseNode) processWithTracing(queryCtx *models.QueryContext, lhs block.Block, rhs block.Block) (block.Block, error) {
```

Comment:
Correct me if I'm wrong, but this will show up on the span as two function calls, one of which will be pretty much instantaneous, right? Should be fine, and probably not worth overengineering here to avoid that case (like starting the span only when both LHS and RHS satisfied), but just trying to get an idea of how it will look :)

Reply:
That being said, it might be necessary to add a parent span to `NewScalarOp`, since there is a potential (highly unlikely, but still) scenario where you have a binary function between a series and a scalar, e.g. `up + 1`, and somehow the `up` block resolves first, which may cut off your trace early if I'm understanding the flow correctly?

Reply:
> Correct me if I'm wrong, but this will show up on the span as two function calls, one of which will be pretty much instantaneous, right?

Ah yeah good question. It actually shows up as a single trace, since I am in fact starting the span only when both LHS and RHS are satisfied.

Here's a sample:

![screen shot 2019-02-15 at 3 21 15 pm](https://user-images.githubusercontent.com/830218/52882449-b6b73b00-3135-11e9-9092-44acc69f01eb.png)

Reply:
Nice!